### ISSUE TITLE
Keras Mixture of MLP Experts

### ISSUE THREAD

**Issue by RezaRB** [fchollet/keras#4085]
Hi, 
I am new in python and theano, and I have a big question about Keras.
Is it possible to implement MLP mixture of expert methodology in Keras?
Could you please guide me by a simple code in Keras for a binary problem with 2 experts.
It needs to define a cost function like this:

g = Gate.layers[-1].output
o1 = MLP1.layers[-1].output
o2 = MLP2.layers[-1].output

def ME_objective(y_true, y_pred):
    A = g[0] \* T.exp(-0.5 \* T.sqr(y_true – o1))
    B = g[1] \* T.exp(-0.5 \* T.sqr(y_true – o2))
    cost = -T.log((A+B).sum())
    return cost

Thank you.

**Comment by MaryamAkhavan** [comment]
Hi,
Could you impelemnt your code with loss function?
I faced with this problem. If you did, can you help me?
thank you
